apis/k8ssandra/v1alpha1/cassandraconfig_types.go

```diff
@@ -26,29 +26,29 @@ import (
 
 type CassandraConfig struct {
 	// +optional
-	CassandraYaml CassandraYaml `json:"cassandraYaml,omitempty" cass-config:"*:cassandra-yaml;recurse"`
+	CassandraYaml CassandraYaml `json:"cassandraYaml,omitempty" cass-config:"*:cassandra-yaml,dse:*:cassandra-yaml;recurse"`
 
 	// +optional
-	JvmOptions JvmOptions `json:"jvmOptions,omitempty" cass-config:"*:;recurse"`
+	JvmOptions JvmOptions `json:"jvmOptions,omitempty" cass-config:"*:,dse:*:;recurse"`
 
 	// +optional
-	DSEYaml DSEYaml `json:"dseYaml,omitempty" cass-config:"*:dse-yaml;recurse"`
+	DseYaml DseYaml `json:"dseYaml,omitempty" cass-config:"dse:*:dse-yaml;recurse"`
 }
 
 // CassandraYaml defines the contents of the cassandra.yaml file. For more info see:
 // https://cassandra.apache.org/doc/latest/cassandra/configuration/cass_yaml_file.html
 type CassandraYaml struct {
```

Comment:
I see your point. There is probably room for simplification in the tag syntax. The syntax is complex mostly because of `JvmOptions` in fact. But I won't have time today to do this. Can we do this later?

Reply:
OK I managed to add some syntactic sugar for the case where the option exists for all server types AND all versions. It's slightly better indeed. I will also try to change the syntax a bit to be able to specify a common path for 2 different server types. e.g.:

```
cassandra:>=4.x,dse:>=6.8.x:foo/bar
```

Reply:
For posterity, and since my last commit, here is how the syntax works:

If the option is present in all server types and all versions, simply specify the option:

```
cass-config:"allocate_tokens_for_keyspace"
```

If the option is present in C* 4+ and DSE 6.8+:

```
cass-config:">=4.x,dse@>=6.8.x:audit_logging_options"
```

If the options is present in C* 3.11 under a given path, and in C* 4+ and DSE 6.8+ under a different one:

```
cass-config:"^3.11.x:jvm-options/initial_heap_size;>=4.x,dse@>=6.8.x:jvm-server-options/initial_heap_size"
```

Reply:
Thanks for the detailed explanation. Very helpful. This is fantastic :)